src/main.cpp

```diff
@@ -6909,6 +6909,22 @@ void ProcessTxBaseMsg(const CTransactionBase& txBase, CNode* pfrom)
     }
 }
 
+bool static ValidateStream(CNode* pfrom, CDataStream& vRecv, const char* msgTypeName)
+{
+    auto vRecvStreamSz = ReadCompactSize(vRecv);
+    if (vRecvStreamSz > MAX_INV_SZ)
+    {
+        Misbehaving(pfrom->GetId(), 20);
+        return error("message %s size() = %u", msgTypeName, vRecvStreamSz);
+    }
+    else if(vRecvStreamSz * sizeof(CInv)!= vRecv.in_avail()) {
+        Misbehaving(pfrom->GetId(), 20);
+        return error("invalid message payload for %u", vRecvStreamSz);
+    }
+    vRecv.Rewind(GetSizeOfCompactSize(vRecvStreamSz));
+    return true;
```

Comment:
How about this option, which is more compact and uses the return value of `Rewind()`?
```suggestion
    return vRecv.Rewind(GetSizeOfCompactSize(vRecvStreamSz));
```

Reply:
Done

Reply:
Thanks!